src/unix-manager.c

```diff
@@ -728,6 +729,65 @@ static TmEcode UnixManagerListCommand(json_t *cmd,
     SCReturnInt(TM_ECODE_OK);
 }
 
+static TmEcode UnixSocketSetOSInfo(json_t *cmd,
+                               json_t *answer, void *data)
+{
+    SCEnter();
+    json_t *jarg = NULL;
+    const char *ipv4 = NULL;
+    const char *ipv6 = NULL;
+    const char *ostype = NULL;
+    const char *errmsg = NULL;
+
+    jarg = json_object_get(cmd, "ostype");
+    if (!json_is_string(jarg)) {
+        errmsg = "ostype is not a string";
+        goto error;
+    }
+    ostype = json_string_value(jarg);
+
+    jarg = json_object_get(cmd, "ipv4");
+    if (!json_is_string(jarg)) {
+        errmsg= "ipv4 is not a string";
+        goto error;
+    }
+    ipv4 = json_string_value(jarg);
+
+    if (!ipv4) {
+        jarg = json_object_get(cmd, "ipv6");
+        if (!json_is_string(jarg)) {
+            errmsg= "ipv6 is not a string";
+            goto error;
+        }
+        ipv6 = json_string_value(jarg);
+    }
+
+    if ( ipv4 ) {
+        SCLogInfo("Set host %s to OS %s", ipv4, ostype);
+        if (SCHInfoAddHostOSInfo(ostype, ipv4, 1) < 0) {
```

Comment:
sadly, this function is not thread safe. We can have multiple readers of the data structure storing the host os info. Adding a single mutex wouldn't work either as that is too much of a contention point. So this will need some more thought.

Reply:
What would be the better, spinlocks or build an RCU structure around the radix tree ?

Reply:
As discussed during SuriCon brainstorm, I think the easiest way would be to use the host table. It would only support single IP-addresses though.